system_tests/espresso_e2e_test.go

```diff
@@ -344,49 +159,35 @@ func waitForL1Node(t *testing.T, ctx context.Context) error {
 	})
 }
 
-// We run one L1 node, two L2 nodes and the espresso containers in this function.
-func runNodes(ctx context.Context, t *testing.T) (*NodeBuilder, *TestClient, *BlockchainTestInfo, func()) {
-
-	cleanValNode := createValidationNode(ctx, t, false)
+func TestEspressoE2E(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-	builder, cleanup := createL1ValidatorPosterNode(ctx, t, hotShotUrl)
+	builder, cleanup := createL1AndL2Node(ctx, t)
```

Comment:
I am thinking if we should use this simple version to run the E2E test. I think it is okay, but the complicated version is actually helping us discover some unknown issues(I agree this is annoying and incomprehensible).
Like we run 2 nodes and one is fed by the other, and this one also is a validator. In this case, we can detect some `reorg` happens in this node because this node gets messages from different sources. And that's why we did some modification on the `tx_streamer`

Reply:
Feel free to merge this PR. I think running an L1 and L2 node actually can reveal more problems.